server/services/content-types.js

```diff
@@ -146,38 +215,102 @@ const manageRelations = async (newData, uid, oldVersionId, model) => {
-    const prevRel = previousVersion[rel];
-    if (prevRel) {
-      const newDataRel = newData[rel];
-      const newDataConnects = newDataRel.connect;
-      const newDataDisconnects = newDataRel.disconnect;
-      //Connect relations from previous version but only if they were not changed by user in the current version.
-      const idsHandledByEditor = [
-        ...newDataConnects.map((conn) => conn.id),
-        ...newDataDisconnects.map((conn) => conn.id),
-      ];
-      const mergedConnects = [...newDataConnects];
-      const prevRelIds = Array.isArray(prevRel)
-        ? prevRel.map((rel) => rel.id)
-        : [prevRel.id];
-      prevRelIds.forEach((pid) => {
-        if (!idsHandledByEditor.includes(pid)) {
-          mergedConnects.push(pid);
+
+  const mergeConnections = (newDataRel, prevRel) => {
+    const newDataConnects = newDataRel.connect;
+    const newDataDisconnects = newDataRel.disconnect;
+
+    //Connect relations from previous version but only if they were not changed by user in the current version.
+    const idsHandledByEditor = [
+      ...newDataConnects.map((conn) => conn.id),
+      ...newDataDisconnects.map((conn) => conn.id),
+    ];
+
+    const mergedConnects = [...newDataConnects];
+
+    const prevRelIds = Array.isArray(prevRel)
+      ? prevRel.map((rel) => rel.id)
+      : [prevRel.id];
+    prevRelIds.forEach((pid) => {
+      if (!idsHandledByEditor.includes(pid)) {
+        mergedConnects.push({ id: pid });
+      }
+    });
+
+    return mergedConnects;
+  };
+
+  const updateRelations = (updateableRels, previous, parent = null) => {
+    updateableRels.forEach((relation) => {
+      if (typeof relation === "string") {
+        if (previous) {
+          if (!_.isArray(previous)) {
+            const prevRel = previous[relation] ?? undefined;
+
+            if (prevRel) {
+              const newDataRel = parent
+                ? newData[parent][relation]
+                : newData[relation];
+
+              const mergedConnects = mergeConnections(newDataRel, prevRel);
+
+              if (!connects[parent]) {
+                connects[parent] = [
+                  {
+                    ...newData[parent],
+                    ..._.pickBy(previousVersion[parent]),
+                  },
+                ];
+              }
+              if (!parent) {
+                connects[relation] = { connect: mergedConnects };
+                return;
+              }
+              connects[parent][relation] = { connect: mergedConnects };
+            }
+            return;
+          }
+
+          previous.forEach((prev, i) => {
+            const prevRel = prev[relation] ?? undefined;
+
+            if (prevRel) {
+              const newDataRel = newData[parent][i][relation];
+              const mergedConnects = mergeConnections(newDataRel, prevRel);
+
+              if (!connects[parent]) {
+                connects[parent] = [
+                  {
+                    ...newData[parent][i],
```

Comment:
shouldn't this be ```...newData[parent]``` instead of ```...newData[parent][i]```, all my other dynamiczone elements are deleted and only the relation one is left if its ```...newData[parent][i]```, using ```...newData[parent]``` fixes this issue

Reply:
@henrik561 it might be. We discarded development of this plugin because it took us too much time considering there will be versioning feature in v5